resolwe_bio/processes/import_data/seq_reads.yml

```diff
@@ -218,3 +218,248 @@
       re-save-file-list fastqc_archive2 "${RW_FASTQC[@]}"
       re-save-list fastqc_url "${FW_FASTQC_URL[@]}"
       re-save-list fastqc_url2 "${RW_FASTQC_URL[@]}"
+
+- slug: basespace-fastq-single
+  name: BaseSpace Reads (single-end)
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: resolwebio/legacy:latest
+    resources:
+      network: true
+      secrets: true
+  data_name: 'BaseSpace {{ file_ids.0|default("?") }}'
+  version: 0.0.1
+  type: data:reads:fastq:single
+  flow_collection: sample
+  category: upload
+  persistence: RAW
+  description: >
+    Import NGS reads in FASTQ format from BaseSpace.
+  input:
+    - name: file_ids
+      label: BaseSpace NGS reads (FASTQ) file IDs
+      type: list:basic:string
+    - name: access_token_secret
+      label: BaseSpace access token
+      type: basic:secret
+      description: >
+        BaseSpace access token secret handle needed to download the files.
+  output:
+    - name: fastq
+      label: Reads file
+      type: list:basic:file
+    - name: fastqc_url
+      label: Quality control with FastQC
+      type: list:basic:file:html
+    - name: fastqc_archive
+      label: Download FastQC archive
+      type: list:basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      {% for file_id in file_ids %}
+        BASESPACE_FILE_ARGUMENT+='--file-id {{file_id}} '
+      {% endfor %}
+
+      SECRET_PATH=/secrets/{{ access_token_secret.handle }}
+      basespace_download.py ${BASESPACE_FILE_ARGUMENT} --access-token-secret-path ${SECRET_PATH}
+      re-checkrc "Failed downloading files from BaseSpace."
+
+      gzip -dk *.gz
+
+      FASTQ=(*.fastq)
```

Comment:
Reused from https://github.com/genialis/resolwe-bio/blob/23e9deb50a8a972b544187a92b0180488b46a7b8/resolwe_bio/processes/import_data/seq_reads.yml#L48

Maybe it would be a good idea to encapsulate this functionality. cc @jkokosar

Reply:
As mentioned in my security-related comment above, having this post-processing step as a separate process, encapsulating the two steps in a workflow would solve this duplication issue.